### ISSUE TITLE
[dev_qarray] Add by zero error for sparse QArrays

### ISSUE THREAD

**Issue by HarshBabla99** [dynamiqs/dynamiqs#805]
In the `dev_qarray` branch, adding by a zero gives an error. Here is a MWE: 

```python
import dynamiqs as dq
z = dq.zero(5)
a = dq.destroy(5)
z+a
```

Here is the traceback:

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[1], line 6
      3 z = dq.zero(5)
      4 a = dq.destroy(5)
----> 6 z+a

File .../python3.12/site-packages/dynamiqs/qarrays/sparsedia_qarray.py:337, in SparseDIAQArray.__add__(self, other)
    335     return self._add_scalar(other)
    336 elif isinstance(other, SparseDIAQArray):
--> 337     return self._add_sparse(other)
    338 elif isqarraylike(other):
    339     warnings.warn(warning_dense_addition, stacklevel=2)

    [... skipping hidden 1 frame]

File .../python3.12/site-packages/dynamiqs/qarrays/sparsedia_qarray.py:367, in SparseDIAQArray._add_sparse(self, other)
    364         other_diag = other.diags[..., other.offsets.index(offset), :]
    365         out_diags = out_diags.at[..., i, :].add(other_diag)
--> 367 return SparseDIAQArray(self.dims, _numpy_to_tuple(out_offsets), out_diags)

File .../python3.12/site-packages/equinox/_module.py:573, in _ModuleMeta.__call__(cls, *args, **kwargs)
    571         pass
    572     else:
--> 573         check(self)
    574 return self

File .../python3.12/site-packages/dynamiqs/qarrays/sparsedia_qarray.py:96, in SparseDIAQArray.__check_init__(self)
     93 # check that diagonals contain zeros outside the bounds of the matrix
     94 for i, offset in enumerate(self.offsets):
     95     if (offset < 0 and jnp.any(self.diags[..., i, offset:] != 0)) or (
---> 96         offset > 0 and jnp.any(self.diags[..., i, :offset] != 0)
     97     ):
     98         raise ValueError(
     99             'Diagonals of a `SparseDIAQArray` must contain zeros outside the '
    100             'matrix bounds.'
    101         )

File .../python3.12/site-packages/jax/_src/array.py:336, in ArrayImpl.__getitem__(self, idx)
    333     else:
    334       return out
--> 336 return lax_numpy._rewriting_take(self, idx)

File .../python3.12/site-packages/jax/_src/numpy/lax_numpy.py:6594, in _rewriting_take(arr, idx, indices_are_sorted, unique_indices, mode, fill_value)
   6591       return lax.dynamic_index_in_dim(arr, idx, keepdims=False)
   6593 treedef, static_idx, dynamic_idx = _split_index_for_jit(idx, arr.shape)
-> 6594 return _gather(arr, treedef, static_idx, dynamic_idx, indices_are_sorted,
   6595                unique_indices, mode, fill_value)

File .../python3.12/site-packages/jax/_src/numpy/lax_numpy.py:6603, in _gather(arr, treedef, static_idx, dynamic_idx, indices_are_sorted, unique_indices, mode, fill_value)
   6600 def _gather(arr, treedef, static_idx, dynamic_idx, indices_are_sorted,
   6601             unique_indices, mode, fill_value):
   6602   idx = _merge_static_and_dynamic_indices(treedef, static_idx, dynamic_idx)
-> 6603   indexer = _index_to_gather(shape(arr), idx)  # shared with _scatter_update
   6604   y = arr
   6606   if fill_value is not None:

File .../python3.12/site-packages/jax/_src/numpy/lax_numpy.py:6848, in _index_to_gather(x_shape, idx, normalize_indices)
   6839   msg = ("Array slice indices must have static start/stop/step to be used "
   6840          "with NumPy indexing syntax. "
   6841          f"Found slice({i.start}, {i.stop}, {i.step}). "
   (...)
   6844          "dynamic_update_slice (JAX does not support dynamically sized "
   6845          "arrays within JIT compiled functions).")
   6846   raise IndexError(msg)
-> 6848 start, step, slice_size = _preprocess_slice(i, x_shape[x_axis])
   6849 slice_shape.append(slice_size)
   6851 if core.definitely_equal(step, 1):
   6852   # Avoid generating trivial gather (an optimization)

File .../python3.12/site-packages/jax/_src/numpy/lax_numpy.py:7099, in _preprocess_slice(s, axis_size)
   7097   stop = axis_size if step_gt_0 else -1
   7098 else:
-> 7099   stop = clamp_index(convert_to_index(s.stop), "stop")
   7101 gap = step if step_gt_0 else - step
   7102 distance = (stop - start) if step_gt_0 else (start - stop)

File .../python3.12/site-packages/jax/_src/numpy/lax_numpy.py:7082, in _preprocess_slice.<locals>.clamp_index(i, which)
   7080 if i_ge_0:
   7081   if step_gt_0:
-> 7082     return core.min_dim(axis_size, i)
   7083   else:
   7084     return core.min_dim(axis_size - 1, i)

File .../python3.12/site-packages/jax/_src/core.py:2194, in min_dim(d1, d2)
   2192   return min(d1, d2)
   2193 if d1_is_constant:
-> 2194   return d2.rmin(d1)  # type: ignore[union-attr]
   2195 else:
   2196   return d1.min(d2)

AttributeError: 'float' object has no attribute 'rmin'
```

**Comment by gautierronan** [comment]
Good catch! Fixed in https://github.com/dynamiqs/dynamiqs/pull/807.

**Comment by pierreguilmin** [comment]
Fixed by #807 merged in #812.
